pmd-core/src/main/java/net/sourceforge/pmd/cpd/CpdAnalysis.java

```diff
@@ -196,7 +196,7 @@ public void performAnalysis(Consumer<CPDReport> consumer) {
             LOGGER.debug("Running match algorithm on {} files...", sourceManager.size());
             MatchAlgorithm matchAlgorithm = new MatchAlgorithm(tokens, configuration.getMinimumTileSize());
             List<Match> matches = matchAlgorithm.findMatches(listener, sourceManager);
-            tokens = null; // NOPMD null it out before rendering
+            tokens = null; // null it out before rendering, might help gc
```

Comment:
@oowekyala Is this comment correct or does it not make sense at all? If there is no reason for this useless statement, we should actually remove it...

Reply:
Then let's remove it. Maybe we can restructure the code, so that the scope of "tokens" is as small as possible.

Reply:
I've refactored CpdAnalysis a bit, please take a look.